src/sagemaker/serve/utils/hardware_detector.py

```diff
@@ -0,0 +1,103 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+"""Utilites for identifying and analyzing instance gpu hardware"""
+from __future__ import absolute_import
+
+import logging
+
+from botocore.exceptions import ClientError
+
+from sagemaker import Session
+
+logger = logging.getLogger(__name__)
+
+
+fallback_gpu_resource_mapping = {
+    "ml.p5.48xlarge": {"Count": 8, "TotalGpuMemoryInMiB": 655360},
+    "ml.p4d.24xlarge": {"Count": 8, "TotalGpuMemoryInMiB": 327680},
+    "ml.p3.2xlarge": {"Count": 1, "TotalGpuMemoryInMiB": 16384},
+    "ml.p3.8xlarge": {"Count": 4, "TotalGpuMemoryInMiB": 65536},
+    "ml.p3.16xlarge": {"Count": 8, "TotalGpuMemoryInMiB": 131072},
+    "ml.p3dn.24xlarge": {"Count": 8, "TotalGpuMemoryInMiB": 262144},
+    "ml.p2.xlarge": {"Count": 1, "TotalGpuMemoryInMiB": 12288},
+    "ml.p2.8xlarge": {"Count": 8, "TotalGpuMemoryInMiB": 98304},
+    "ml.p2.16xlarge": {"Count": 16, "TotalGpuMemoryInMiB": 196608},
```

Comment:
out of curiosity, what's the source of truth for this? I didn't see gpu mem info here https://aws.amazon.com/ec2/instance-types/p2/

Reply:
To get the mem, please run command like 
`aws ec2 describe-instance-types --instance-types g5.48xlarge`

Reply:
gotcha, then if this can be retrieved from describe-instance-types, do we still need to store them as constants? Or i misunderstand the functionality of this fallback info. Imo, this is what we rely on when certain instance types are not retrievable from ec2.

Reply:
See here https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/serve/utils/local_hardware.py#L71

We can still store them and for non-retrievable instances from ec2 api, the gpu mem is here https://aws.amazon.com/sagemaker/pricing/